SourceKitStressTester/Sources/StressTester/SourceKitDocument.swift

```diff
@@ -445,20 +435,42 @@ class SourceKitDocument {
     }
   }
 
-  private func sendWithTimeout(_ request: SourceKitdRequest, info: RequestInfo, reopenDocumentIfNeeded: Bool = true) throws -> SourceKitdResponse {
+  /// Send the `request` synchronously, timing out after
+  /// `SOURCEKIT_REQUEST_TIMEOUT`.
+  /// Afterwards, the response is checked, whether it contains an error. If
+  /// either `throwIfInvalid` or `validateResponse` throws an error, it gets
+  /// re-thrown by this error.
+  /// If the request took longer than half the allowed time but the previous
+  /// verifications pass, a `SourceKitError.softTimeout` will be thrown.
```

Comment:
Could use some re-wording, eg. something like
```
An error is thrown if either the response is invalid (see `throwIfInvalid`) or `validateResponse` throws. If the request took longer than half the time allowed but no other error is thrown, a `SourceKitError.softTimeout` will be thrown.
```

Reply:
Thanks, that’s definitely better.